internal/resources/aws/fsx_openzfs_file_system.go

```diff
@@ -0,0 +1,147 @@
+package aws
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+
+	"github.com/shopspring/decimal"
+)
+
+type FSxOpenZFSFileSystem struct {
+	Address             string
+	StorageType         string
+	ThroughputCapacity  int64
+	ProvisionedIOPS     int64
+	ProvisionedIOPSMode string
+	StorageCapacityGB   int64
+	Region              string
+	DeploymentType      string
+	DataCompression     string
+	CompressionSavings  *int64   `infracost_usage:"compression_savings"`
```

Comment:
Ok, with the naming.
I made it an int because in the pricing calculator you can only enter integer values as percentage.
Ha, another question just popped up ... do we need to do some range check for the percentage, i.e. 1-99 % or so?

Reply:
@aliscott Renaming done with 9f10eff

Reply:
> I made it an int because in the pricing calculator you can only enter integer values as percentage.

The reason for making it a float would be for the future when we pull the actual usage from the users' CUR and show the estimated usage based on it. To match the numbers on the CUR exactly. we will probably need this to be a float.

> Ha, another question just popped up ... do we need to do some range check for the percentage, i.e. 1-99 % or so?

Good point. I think we could do something like `min(max(0, CompressionSavingsPercent), 100)` to ensure it's in that range, but we don't have any mechanism for showing validation errors to the user for this yet.

Reply:
> The reason for making it a float would be for the future when we pull the actual usage from the users' CUR and show the estimated usage based on it. To match the numbers on the CUR exactly. we will probably need this to be a float.

Understood. Fixed with 4d9a2d4.

> Good point. I think we could do something like `min(max(0, CompressionSavingsPercent), 100)` to ensure it's in that range, but we don't have any mechanism for showing validation errors to the user for this yet.

Ok, since there's no mechanism for showing such validation errors I would rather go to not modify the given usage value. I left it now as it is (without min/max 0-100). This follows now one of my favourite patterns, when giving values out of bounds to some function: sh*t in means sh*t out.
I tested it with negative percent values or >100 percent and it doesn't break, which is the most important for me, and still gives you the mathematical correct values: negative values will give you an output where you pay more and having a compression rate better then 100% will give you negative spends 😉

Reply:
👍 yeah that sounds good. Thank you!